src/xpk/core/docker_manager.py

```diff
@@ -240,13 +193,13 @@ def run_command(
     except APIError as e:
       xpk_print(f"Deploying cluster toolkit failed due to {e.explanation}")
       xpk_exit(DockerRunCommandExitCode)
-
-  def _make_upload_directory(self, name: str) -> str:
-    target_path = os.path.join(self.working_dir, upload_dir, name)
-    target_dir = os.path.join(self.working_dir, upload_dir)
-    if not os.path.exists(target_dir):
-      os.mkdir(target_dir)
-    return target_path
+    finally:
```

Comment:
Why we don't trust docker api?

Reply:
when detached run, I saw it wasn't removed in time before the next run. it was for when we did not have unique name though

Reply:
ok, never saw this in tests.

Reply:
Did you test detach?